core-bundle/src/Twig/Compat/ArrayValueHolder.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Twig\Compat;
+
+final class ArrayValueHolder extends \ArrayIterator implements SafeHTMLValueHolderInterface
+{
+    /**
+     * @var string
+     */
+    private $valueName;
+
+    /**
+     * @internal
```

Comment:
Shouldn't this be at class level?

Reply:
Imho it can't be class level because you need to consume this class (but not instantiate it).

Reply:
But how is someone supposed to use the class if the constructor is internal?

Reply:
Your IDE at least marks it when you create an object 🙃

Reply:
I think it's the same with every class that has a factory. The constructor isn't part of the BC promise, the factory method constructing it, however, is. In this case you would use `ProxyFactory::createValueHolder(…)` instead of `new ArrayValueHolder(…)`.

We've got the same in place for the studio classes for instance. :slightly_smiling_face:

Reply:
I see. Maybe we should add a comment: `@internal Do not instantiate this class in your code; use ProxyFactory::createValueHolder() instead`